clusterclient.go

```diff
@@ -0,0 +1,120 @@
+package kafka
+
+import (
+	"context"
+	"fmt"
+)
+
+// ClusterClient provides a high-level API for miscellaneous (i.e. non consume or produce) Kafka cluster operations
+type ClusterClient struct {
+	brokers []string
+	dialer  *Dialer
+}
+
+// NewClusterClient creates and returns a *ClusterClient with a []string of bootstrap
+// brokers for connecting to the cluster and the DefaultDialer
+func NewClusterClient(brokers []string) *ClusterClient {
+	return &ClusterClient{
+		brokers: brokers,
+		dialer:  DefaultDialer,
+	}
+}
+
+// NewClusterClient creates and returns a *ClusterClient with a []string of bootstrap
+// brokers for connecting to the cluster and a user supplied Dialer
+func NewClusterClientWith(brokers []string, d *Dialer) *ClusterClient {
+	return &ClusterClient{
+		brokers: brokers,
+		dialer:  d,
+	}
+}
+
+// ConsumerOffsets returns a map[int]int64 of partition to committed offset for a consumer group id and topic
+func (cc *ClusterClient) ConsumerOffsets(ctx context.Context, groupId string, topic string) (map[int]int64, error) {
+	address, err := cc.lookupCoordinator(groupId)
+	if err != nil {
+		return nil, err
+	}
+
+	conn, err := cc.coordinator(ctx, address)
+	if err != nil {
+		return nil, err
+	}
+
+	defer conn.Close()
+	partitions, err := conn.ReadPartitions(topic)
+	if err != nil {
+		return nil, err
+	}
+
+	var parts []int32
+	for _, p := range partitions {
+		parts = append(parts, int32(p.ID))
+	}
+
+	offsets, err := conn.offsetFetch(offsetFetchRequestV1{
+		GroupID: groupId,
+		Topics: []offsetFetchRequestV1Topic{
+			{
+				Topic:      topic,
+				Partitions: parts,
+			},
+		},
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	offsetsByPartition := map[int]int64{}
+	for _, pr := range offsets.Responses[0].PartitionResponses {
```

Comment:
Is it possible to get no responses and no errors?

Reply:
Addressed in latest